### PR TITLE
Add periodic job to test Kubernetes modules with minimum Go version

### PR DESCRIPTION
**Pull request by BenTheElder**
AI assisted. human reviewed. based on the existing gotip job.

/hold
/cc @dims 

Context is: https://github.com/kubernetes/kubernetes/issues/130799

### CODE REVIEW

File: config/jobs/kubernetes/sig-arch/kubernetes-code-organization.yaml [hunk 1]
```diff
@@ -420,6 +420,54 @@ periodics:
             cpu: 4
             memory: "8Gi"
 
+- interval: 24h
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-arch/kubernetes-code-organization.yaml:423]: only 24h to start. we can raise it later and promote to release informing perhaps

File: config/jobs/kubernetes/sig-arch/kubernetes-code-organization.yaml [hunk 2]
```diff
@@ -420,6 +420,54 @@ periodics:
...
+  cluster: k8s-infra-prow-build
+  name: ci-kubernetes-test-modules-with-minimum-go
+  annotations:
+    testgrid-dashboards: sig-arch-code-organization
+    testgrid-tab-name: test-modules-with-minimum-go
+    description: Tests staging modules with the minimum Go version advertised in each module's go.mod, the way an external consumer would.
+    testgrid-alert-email: davanum@gmail.com, bentheelder@google.com
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-arch/kubernetes-code-organization.yaml:430]: @dims I kept you from the other job because you've been shepherding go version break jobs, but I'll drop if you'd rather not. Maybe later we add a mailinglist.

**Comment by BenTheElder** [comment]
Uhhh
/test
